### PR TITLE
Optimus Post Processing

### PR DESCRIPTION
**Pull request by jessicaway**
Adds a wdl, tasks, scripts, and docker to combine loom outputs into a single roject-level loom. Also uses the json files corresponding to intermediate loom files to create the adapter json files for the combined matix loom.

### CODE REVIEW

**Review by khajoue2 — APPROVED** [review]

File: dockers/skylab/HCA_post_processing/optimus_HCA_loom_merge.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+                        required=True,
+                        help="Path to output loom file")
+    args = parser.parse_args()
+
+    loom_file_list = args.input_loom_files
+    library = " ".join(args.library)
+    species = " ".join(args.species)
+    organ = " ".join(args.organ)
+
+    attr_dict = {"library_costruction_method": library,
```
Comment (ekiernan) [inline dockers/skylab/HCA_post_processing/optimus_HCA_loom_merge.py:41]: costruction --> construction?

Comment (ekiernan) [reply dockers/skylab/HCA_post_processing/optimus_HCA_loom_merge.py]: Update looks good! Thank you!

**Review by ekiernan — APPROVED** [review]

File: dockers/skylab/HCA_post_processing/HCA_create_adapter_json.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    # named with a UUID) to ensure it is the subworkflow id
+    url = analysis_output_path.rsplit('/', 1)[0]
+    uuid_regex = r"([a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12})"
+    workflow_id = re.findall(uuid_regex, url)[-1]
+    print('Got analysis workflow UUID: {0}'.format(workflow_id))
+    return workflow_id
+
+
+def main():
+    description = """Add metadata into a Loom file as column attributes"""
```
Comment (kishorikonwar) [inline dockers/skylab/HCA_post_processing/HCA_create_adapter_json.py:33]: Create JSON files for HCA DCP2 MVP.

File: dockers/skylab/HCA_post_processing/HCA_create_adapter_json.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    size = args.size
+    staging_bucket = args.staging_bucket
+    with open(args.inputs_json, "r") as i:
+        inputs_dict = json.load(i)  # this should be a list of dictionaries
+        inputs = inputs_dict['inputs']
+    with open(args.protocols_json, "r") as p:
+        protocols_dict = json.load(p)  # this should be a list of dictionaries
+        protocols = protocols_dict['protocols']
+
+    # Generate additional data from agrs
```
Comment (kishorikonwar) [inline dockers/skylab/HCA_post_processing/HCA_create_adapter_json.py:87]: agrs --> args

File: dockers/skylab/HCA_post_processing/HCA_create_adapter_json.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+                             "outputs": [{
+                                         "output_id": matrix_file_uuid,
+                                         "output_type": "analysis_file"
+                                         }],
+                             "protocols": protocols,
+                             "link_type": "process_link"
+                             }]
+                  }
+
+    # filenames for staging dierctories
```
Comment (kishorikonwar) [inline dockers/skylab/HCA_post_processing/HCA_create_adapter_json.py:132]: dierctories --> directories

**Review by kishorikonwar — APPROVED** [review]

**Pull request merged**